BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -164,7 +172,11 @@ public override void OnStart(StartState state)
                 part.force_activate();
                 sourcevessel = vessel;
                 var MF = VesselModuleRegistry.GetModule<MissileFire>(vessel, true);
-                if (MF != null) sourcevessel = MF.vessel; // grab the vessel the Weapon manager is on at start
+                if (MF != null)
+                {
+                    sourcevessel = MF.vessel;
+                    sourceteam = MF.Team; // grab the vessel the Weapon manager is on at start
```

Comment:
This doesn't cover what happens when the teams are reassigned. `SourceVesselTeam` needs to get updated whenever the `sourcevessel` changes teams. Also, `sourceteam` seems a bit redundant since it's never used outside of setting `SourceVesselTeam`.

Reply:
Original thought was something along the lines of why would missiles update IFF if teams flipped for whatever reason after they were in the air, but fair. I'll move the sourceTeam = to the start of DetonateIfpossible, as that's the only time we need the team, it's not going to be changing in the next milisecond, and this way we only need to run it once without anything in the background to update it if it changes.